### PR TITLE
Implement SQLAlchemy ORM and resources(APIs)

### PR DESCRIPTION
**Pull request by patilsuraj767**
Aim of this PR is to create the DB model to handle/store performance data and to make it available for the frontend.

### CODE REVIEW

**Review by kgaikwad — CHANGES_REQUESTED** [review]
Thank you @patilsuraj767.
Overall looks good to me.
Going through SQLAlchemy documentation to get better understanding of auto generated files & syntax. 
Added few inline comments.

File: ros/resources/hosts.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                {
+                    'fqdn': 'machine1.local.company.com',
+                    'display_name': 'machine1-rhel_test123',
+                    'id': '12345-57575757',
+                    'account': '12345',
+                    'vm_uuid': '12345a1',
+                    'state': 'Oversized',
+                    'recommendation_count': 4,
+                    'organization_id': 1,
+                    'performance_profile': {
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply ros/resources/hosts.py:20]: Ok, I will change the parent key to `performance_score` in dummy_data. Child keys can remain the same.

File: ros/app.py [hunk 2]
```diff
@@ -1,68 +1,24 @@
...
-
-
-@app.route('/api/systems')
-def list_systems():
-    """Function returns list of systems."""
-    return jsonify(DUMMY_SYSTEMS)
-
+# Routes
+api.add_resource(Status, '/api/status')
+api.add_resource(Hosts, '/api/hosts')
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply ros/app.py:20]: Ok, I will change the API to `/api/systems`

File: migrations/versions/331c56fed399_.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('performance_profile',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('host_id', postgresql.UUID(as_uuid=True), nullable=False),
+    sa.Column('performance_record', postgresql.JSONB(astext_type=sa.Text()), nullable=True),
+    sa.Column('performance_score', postgresql.JSONB(astext_type=sa.Text()), nullable=True),
+    sa.Column('report_date', sa.DateTime(), nullable=True),
+    sa.PrimaryKeyConstraint('id'),
+    sa.UniqueConstraint('host_id')
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply migrations/versions/331c56fed399_.py:28]: Yes, it should be `host_id + report_date` I will add unique constrain on `report_date`

File: ros/resources/hosts.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kgaikwad) [reply ros/resources/hosts.py:20]: +1, Also could you change `facts` key to `performance_record`?

[the remark this answers is not in the document]
Comment (patilsuraj767) [reply ros/resources/hosts.py:20]: I am not sure if we are going to store facts like `provider, instance_type` etc in our ROS DB. I think we should fetch it from host-inventory when the frontend calls the backend. Also do we really need to expose the `performance_record` or performance_score in enough?

File: migrations/versions/5c21ad23b0f7_.py [hunk 4]
```diff
@@ -20,12 +20,13 @@ def upgrade():
...
     sa.Column('id', sa.Integer(), nullable=False),
-    sa.Column('host_id', postgresql.UUID(as_uuid=True), nullable=False),
+    sa.Column('inventory_id', postgresql.UUID(as_uuid=True), nullable=False),
     sa.Column('performance_record', postgresql.JSONB(astext_type=sa.Text()), nullable=True),
     sa.Column('performance_score', postgresql.JSONB(astext_type=sa.Text()), nullable=True),
-    sa.Column('report_date', sa.DateTime(), nullable=True),
+    sa.Column('report_date', sa.Date(), nullable=True),
     sa.PrimaryKeyConstraint('id'),
-    sa.UniqueConstraint('host_id')
+    sa.UniqueConstraint('inventory_id'),
```
Comment (kgaikwad) [inline migrations/versions/5c21ad23b0f7_.py:28]: Can't we define a single unique constraint on these two columns?
For now if we can handle logic of host update on processor level then I think it is better to declare combined key? 
what do you think?

File: ros/resources/hosts.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kgaikwad) [reply ros/resources/hosts.py]: After integration with host-inventory, fetched data from inventory that is required on front-end will need to be expose in API. I think for now, I am consuming details using `facts` key.

File: migrations/versions/5c21ad23b0f7_.py (hunk 4 above)
Comment (patilsuraj767) [reply migrations/versions/5c21ad23b0f7_.py:28]: You are right.  Obviously, Report_date cannot be unique. A unique constraint should be on two columns `inventory_id` and `report_date`.

Changing the code so DB schema will look like below. 
```
postgres=# \d performance_profile;
                                  Table "public.performance_profile"
       Column       |  Type   | Collation | Nullable |                     Default                     
--------------------+---------+-----------+----------+-------------------------------------------------
 id                 | integer |           | not null | nextval('performance_profile_id_seq'::regclass)
 inventory_id       | uuid    |           | not null | 
 performance_record | jsonb   |           |          | 
 performance_score  | jsonb   |           |          | 
 report_date        | date    |           |          | 
Indexes:
    "performance_profile_pkey" PRIMARY KEY, btree (id)
    "performance_profile_inventory_id_key" UNIQUE CONSTRAINT, btree (inventory_id)
    "performance_profile_inventory_id_report_date_key" UNIQUE CONSTRAINT, btree (inventory_id, report_date)
```

**Review by kgaikwad — CHANGES_REQUESTED** [review]

File: ros/app.py [hunk 5]
```diff
@@ -1,68 +1,24 @@
...
-
-@app.route('/api/systems')
-def list_systems():
-    """Function returns list of systems."""
-    return jsonify(DUMMY_SYSTEMS)
-
+# Routes
+api.add_resource(Status, '/api/status')
+api.add_resource(Hosts, '/api/systems')
+api.add_resource(HostDetails, '/api/system/<host_id>')
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply ros/app.py:21]: The API call is `/api/system/` so param should be system_id or host_id, inventory_id seems to be little misleading.

File: migrations/versions/5e5c57ed6627_.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('performance_profile',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('inventory_id', postgresql.UUID(as_uuid=True), nullable=False),
+    sa.Column('performance_record', postgresql.JSONB(astext_type=sa.Text()), nullable=True),
+    sa.Column('performance_score', postgresql.JSONB(astext_type=sa.Text()), nullable=True),
+    sa.Column('report_date', sa.Date(), nullable=True),
+    sa.PrimaryKeyConstraint('id'),
+    sa.UniqueConstraint('inventory_id'),
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply migrations/versions/5e5c57ed6627_.py]: Removed the constraint from `inventory_id`

File: ros/app.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (kgaikwad) [reply ros/app.py:21]: Okies.
What about API path `/api/systems/{whatever_param_id_name}?
Example - in Rest API for GET user path is `/users/{user_id}`

**Review by kgaikwad — APPROVED** [review]
Thank you @patilsuraj767 for patience.
LGTM :+1: 
No further comments. Merging it.

**Pull request merged**